### PR TITLE
Changed internal temporal representations to Instant/Duration to support sub-second precision.

### PR DESCRIPTION
**Pull request by jyelloz**
This should improve summary information accuracy when working with small time windows. #2

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -556,7 +555,7 @@ mod tests {
         assert_eq!(queue.iter().collect::<Vec<_>>(), vec![&1, &5, &2, &50]);
         println!("Same elements + 50: {:?}", queue.iter().collect::<Vec<_>>());
 
-        sleep_secs(2);
+        sleep_secs(1);
```
Comment (jyelloz) [inline src/lib.rs:558]: Over here, maybe switching to a fake clock for the tests can make the tests deterministic but that would increase complexity a bit.

Comment (mrsarm) [reply src/lib.rs:558]: Actually this test was wrong, waiting 2 sec should make the element to expire and your improvements and the change here put in evidence that.

**Review by mrsarm — APPROVED** [review]
Thanks for this PR and sorry for the delay, this is very useful, and I'm going to release a new version of the lib soon.

**Comment by mrsarm** [comment]
Hey @jyelloz , do you think would be useful to add more methods to create a new queue with a duration with a different granularity than seconds? do you need to setup queues with less than 1 second? thinking in another PR before releasing...

**Comment by jyelloz** [comment]
Thanks for merging the code. Using Duration in the constructors clarifies the numerical units, so adding constructors and getters might make it clearer for new users what values they are working with. It also might reduce the amount of type conversions from duration to seconds that a user of the library would have to supply to the sum-queue API assuming the application code is starting with a Duration and wants to create a SumQueue. I don't really see any disadvantages to adding support for supplying a duration to the constructor API or retrieving the duration from the queue.

**Comment by mrsarm** [comment]
Agree, I'm creating a new PR for this.

**Pull request merged**